### DISCUSSION TITLE
DateTimeOffset.Should().ReallyBeTheSameAs....

### DISCUSSION THREAD

**Discussion by gliljas** [fluentassertions/fluentassertions#1608] [Ideas]
Is there a way to, in a specific case, assert that a DateTimeOffset value is exactly the same as some other DateTimeOffset, i.e comparing it with DateTimeOffset.EqualsExact?

I can add And.HaveOffset, but that's not quite as nice.

The BeExactly assertion is already stolen, so if I'll submit a PR for this, what would be a good name for this? BeExactly could be used, since it would be a different method, but that would mean having two quite different assertions with the same name.

**Comment by dennisdoomen** [comment]
How is that different from `Be`?

**Reply by jnyrup** [reply]
I had a look at https://source.dot.net/#System.Private.CoreLib/DateTimeOffset.cs,396 and I learned something new.

`Equals` and `operator ==` (which we use) checks if two `DateTimeOffset`s points to the same point in time, whereas `EqualsExact` checks if the DateTime and Offset part of two instances are equal.

```cs
var a = new DateTimeOffset(new DateTime(2020, 1, 1, 1, 0, 0), TimeSpan.FromHours(1));
var b = new DateTimeOffset(new DateTime(2020, 1, 1, 2, 0, 0), TimeSpan.FromHours(2));
Console.WriteLine(a == b); // True
Console.WriteLine(a.EqualsExact(b)); // False
```

**Reply by eNeRGy164** [reply]
So if you look at https://stackoverflow.com/a/14268167/581675
`Equals` test for the _Instantaneous time_ and `EqualsExact` tests for the _Calendar Time_?

**Reply by gliljas** [reply]
Exactly (pun intended). The normal Equals tests is probably the most common scenario to assert, and changing "Be" to use EqualsExact would probably break many tests, although replacing it with "BeSamePointInTimeAs" has some merit.

**Comment by gliljas** [comment]
PR #1609 added
